test/vaadin-list-mixin.html

```diff
@@ -479,6 +478,142 @@
           expect(list._scrollerElement.scrollTop).to.be.greaterThan(0);
         });
       });
+
+      describe('Multiple', () => {
+        it('should clear selected when multiple=true', () => {
+          list.selected = 3;
+          list.multiple = true;
+          expect(list.selected).to.be.equal(undefined);
+        });
+
+        it('should move selected to selectedValues when multiple=true', () => {
+          list.selected = 3;
+          list.multiple = true;
+          expect(list.selectedValues).to.eql([3]);
+        });
+
+        it('should clear selectedValues when multiple=false', () => {
+          list.multiple = true;
+          list.selectedValues = [3];
+          list.multiple = false;
+          expect(list.selectedValues).to.eql([]);
```

Comment:
Native select keeps the value in `selectedOptions` also in single selection mode. Again, not sure if we should totally follow it.